### ISSUE TITLE
IPCMessageReader/Writer not re-exported from vscode-languageserver/node

### ISSUE THREAD

**Issue by mattacosta** [microsoft/vscode-languageserver-node#646]
Just tried updating one of my extensions to a 7.0.0 preview build and noticed that the `IPCMessageReader` and `IPCMessageWriter` classes are no longer re-exported. Technically, this only affects old extensions built from the early extension samples, and it has an easy workaround, but as far as I know those classes weren't deprecated and there was nothing in the readme.

**Comment by dbaeumer** [comment]
This is caused by the split into common, node and browser. To access the two classes you need to import `vscode-jsonrpc\node` as documented in the readme.

**Comment by mattacosta** [comment]
Yes, as I mentioned the fix is trivial, but this issue is more about consistency and providing an easy upgrade path. 

Currently, `vscode-languageserver-protocol` re-exports all the types from `vscode-jsonrpc` across all three variants. The `vscode-languageserver` module then goes on to only re-exports types from that package in the [common](https://github.com/microsoft/vscode-languageserver-node/blob/340e65f5b7a3663adf11c69e78c68835514ed9dd/client/src/common/api.ts#L6) and [browser](https://github.com/microsoft/vscode-languageserver-node/blob/340e65f5b7a3663adf11c69e78c68835514ed9dd/client/src/browser/main.ts#L10) variants, but not the node variant.

For those upgrading extensions, this is annoying because in the vast majority of cases one can literally replace `vscode-languageserver` with `vscode-languageserver/node` and have things just work. This issue changes the API surface (as in it's not available from the package at all), which is a breaking change in my opinion that goes beyond a simple "stuff has been split" note in the readme.

**Comment by dbaeumer** [comment]
@mattacosta actually you have a very valid point. The node part of the server should also reexport the node part of the protocol. I will reopen the issue to address this.

**Comment by dbaeumer** [comment]
Published new client and server packages.
